### ISSUE TITLE
Wrong execution order with nested Example

### ISSUE THREAD

**Issue by oswaldo** [etorreborre/specs2#446]
The following code prints as expected:

```
@RunWith(classOf[JUnitRunner]) //to be able to run with surefire maven plugin
class SpecsSampleSpec extends Specification with BeforeAfterAll {
  sequential

  override def beforeAll = {
    println("beforeAll")
  }

  override def afterAll = {
    println("afterAll")
  }

  "first" in {
    println("first")
    success
  }

  "second" in {
    println("second")
    true should beTrue
  }

  "third" in {
    println("third")
    ok
  }

}
```

Prints:

>-------------------------------------------------------
> T E S T S
>-------------------------------------------------------
>Running SpecsSampleSpec
>beforeAll
>first
>second
>third
>afterAll
>Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.296 sec

But adding a nested Example breaks the order:

```
@RunWith(classOf[JUnitRunner])
class SpecsSampleSpec extends Specification with BeforeAfterAll {
  sequential

  override def beforeAll = {
    println("beforeAll")
  }

  override def afterAll = {
    println("afterAll")
  }

  "first" in {
    println("first")
    success

    //this breaks the order
    "then" in {
      print("then")
      ok
    }
  }

  "second" in {
    println("second")
    true should beTrue
  }

  "third" in {
    println("third")
    ok
  }

}
```

Prints:

>-------------------------------------------------------
> T E S T S
>-------------------------------------------------------
>first
>Running SpecsSampleSpec
>first
>beforeAll
>thensecond              <-Something was run in parallel?
>third
>afterAll
>Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.295 sec 

Using Java 8, Specs2 3.6.6, Scala 2.11.7, Maven 3.3.3, Surefire 2.18.1 (aparently it does not work with the latest, 2.19)

**Comment by etorreborre** [comment]
This is not really a bug, but not really a feature either :-(. In specs2 you can nest blocks but you can not have an "example" be a part of another "example".

This means that the code
```
println("first")
success
```
is not considered as an example which must be reported as such but just as some code to execute before creating the next block which is the "then" example. The correct form is
```
"first" in {
   "do this" in {
      println("first")
      success
   }

    //this breaks the order
    "then" in {
      print("then")
      ok
    }
  }
```

Note however that there is a possibility to create a ["continuation"](https://etorreborre.github.io/specs2/guide/SPECS2-3.6.6/org.specs2.guide.CreateOnlineSpecifications.html) for a given example. This is interesting because the result of one example may decide on the creation or not of many other ones.

**Comment by oswaldo** [comment]
Fair enough, thanks for the quick response.

I just thought it would be supported as I saw no exception saying otherwise. Maybe that would be a improvement, to throw something with a message like  ```Example "Y" nested in "X" on specification "Z", but example nesting is not supported. Please restructure your spec or use continuations```, if you agree that something that is not supported should throw accordingly. (If it will never be supported is a different question that if so, I would expect some documentation explaining why)

I reported this because if I see a block being executed twice before the method that as far as I understand is to be executed before everything else, with no message stating that that is forbidden, usually I call a bug or at least a missing feature (exceptions giving me a hint on how to proceed).

In any case, thanks for the great library :smile: 

I have restructured the actual test and everything works now.

**Comment by etorreborre** [comment]
I would like this to be checked at compile time but since this is not possible, it would be nice to have a runtime check at least. I'll see what I can do. It will also be better than putting it in the documentation because it is likely to be drowned in the noise.

**Comment by etorreborre** [comment]
I just thought about it. There is unfortunately no way for me to detect this case, even at runtime, because this is some arbitrary successful code executed before the example definition
```
    println("first")
    success
    //this breaks the order
    "then" in {
      print("then")
      ok
    }
```

The only thing I can do is to detect when things go wrong
```
    println("first")
    1 must_== 2
    "then" in {
      print("then")
      ok
    }
```

In that case I can catch the exception and return a meaningful error message.
